test/ibm/test_registration.py

```diff
@@ -46,25 +47,30 @@
 
 # TODO: NamedTemporaryFiles do not support name in Windows
 @skipIf(os.name == 'nt', 'Test not supported in Windows')
-class TestCredentials(IBMQTestCase):
+class TestCredentials(IBMTestCase):
     """Tests for the credential modules."""
 
     def test_load_account_no_credentials(self) -> None:
-        """Test ``load_account()`` with no credentials available."""
+        """Test load account with no credentials available."""
+
         with custom_qiskitrc(), no_envs(CREDENTIAL_ENV_VARS):
-            with self.assertRaises(IBMQAccountError) as context_manager:
-                IBMQ.load_account()
+            with self.assertRaises(IBMProviderError) as context_manager:
+                IBMProvider()
 
-        self.assertIn('No IBM Quantum Experience credentials found',
+        self.assertIn('No IBM Quantum credentials found',
                       str(context_manager.exception))
 
+    # Test not supported in Python 3.6 since patching a classmethod
+    # (in mock_ibm_provider) incorrectly throws
+    # TypeError: 'NonCallableMagicMock' object is not callable
+    # which was fixed in later versions
```

Comment:
Doesn't have to be in this PR, but we can just drop Python 3.6 since it's end of life is the end of this year anyway.

Reply:
Created new issue https://github.com/Qiskit-Partners/qiskit-ibm/issues/86